### ISSUE TITLE
Milestone for 2.0?

### ISSUE THREAD

**Issue by Keruspe** [sunng87/handlebars-rust#267]
Hi,

I see the milestone for 2.0 only has one (closed) issue.
What do you think will be the timeline for the final 2.0 release?

**Comment by sunng87** [comment]
Sorry the milestone was just out of sync. 

In 2.0 release, I plan to:

* shipping break changes that included in yanked 1.2.0 (mostly API refinements)
* refactoring `Output` API to support procedural macro based compile-time template engine, with most of current foundation #252 (no actually progress for now)
* revamping block params support #260 
* use hashbrown to speed up map access (already shipped in betas)

**Comment by Keruspe** [comment]
So, the first and the last points are already shipped in beta, and #252 and #260 are left to do, right?

**Comment by sunng87** [comment]
Yes. If you were interested in picking up #252, I can offer help.

**Comment by Keruspe** [comment]
I don’t really understand what it is about.
 (Fwiw, I use handlebars in some build.rs to generate rust code)

**Comment by sunng87** [comment]
The main goal is to seek to support compile time template, the API can be like:

```rust
#[handlebars(template="user.hbs")]
pub struct User {
  name: String,
  age: u8,
}

let user = User {
  name: "Ning Sun".to_owned(),
  age: 18
};

println!("{:?}". user);
```

**Comment by Keruspe** [comment]
Can't the StringOutput be used for implementing such a proc macro? Or maybe I'm totally missing the point?

**Comment by sunng87** [comment]
this is to expand template logic in compile time and generate a render function. For example, helper `#each` is to be expanded as `for` loop in rust code.

**Comment by sunng87** [comment]
2.0 released and we deferred #252 for future release.
